GoogleForCreators/web-stories-wp
# Design System: Add Radio Button

## Context

Building out design system.
Designs:
- [Radio base](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3447%3A64541)
- [In panel](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3398%3A94450)

## Summary

Adds radio button to the design system.

<img src="https://user-images.githubusercontent.com/22185279/110555609-1d7f6680-80fa-11eb-8af2-d8c482254513.gif" width="500px" />

## Relevant Technical Choices

nothing special

## To-do

- add a `RadioGroup` per ticket requirements. That will come in the next pr.

## User-facing changes

none

## Testing Instructions



1. Go to storybook
2. Visit Design System -> Radio
3. Play with the radio button. Should be keyboard accessible.

Accessibility Spec for Radio buttons:
```
// https://stackoverflow.com/a/14322755
- Focus can move to a radio group via:
    - The Tab key
    - An Access Key or mnemonic targeting the label
    - Activation of the label (through Assistive Technology mechanism)
- The Tab key moves focus between radio button groups and other widgets.
- When focus is on the group and when no radio button is selected:
    - Tab key press moves focus to the first radio button in the group, but does not select the radio button.
    - Shift+Tab key press moves focus to the last radio button in the group, but does not select the radio
      button.
- When focus moves to the group in which a radio button is selected, pressing Tab and Shift+Tab keys move
  focus to the radio button that is checked.
- Up Arrow and Down Arrow keys move focus and selection.
     - Up Arrow key press moves focus and selection forward through the radio buttons in the group. If the
       first radio button has focus, then focus and selection move to the last radio button in the group.
     - Down Arrow key press moves focus and selection backward through the radio buttons in the group.
       If the last radio button has focus, then focus and selection move to the first radio button in the group.
       * Space Bar key press checks the radio button that currently has focus.
- When re-entering the group, focus returns to the point of previous focus (the item with selection set).
- Pressing the Tab key exits the group and moves focus to the next form control.
- Pressing the Shift+Tab keys exits the group and moves focus to the previous form control.
```

### QA






### UAT






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Partially addresses #6501
